extensions/src/ACRE2Arma/arma2ts/CallExt_DllMain.cpp

```diff
@@ -188,33 +187,43 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
         case PIPE_COMMAND_READ: {
             if (readConnected) {
                 DWORD cbRead;
-                DWORD err;
                 BOOL ret;
-                char *value = new char[4096];
+                constexpr size_t read_length = 4097U;
+                char value[read_length]; // Allocate on stack to delegate memory management to compiler,
+                // allows up to 4096 char string (+1 to ensure NUL terminated), like initial version
+
                 //DEBUG("Read from pipe [%d]\n", hPipe);
-                ret = ReadFile(readHandle, (LPVOID)value, 4096, &cbRead, NULL);
-                if ( ! ret) {
-                    err = GetLastError();
-                    if (err == ERROR_BROKEN_PIPE) {
-                        ClosePipe();
-                    }
+                ret = ReadFile(readHandle, (LPVOID)value, sizeof(char) * (read_length - 1U), &cbRead, NULL);
```

Comment:
In addition to deleting `BOOL ret;` in line 190

```suggestion
                const BOOL ret = ReadFile(readHandle, (LPVOID)value, sizeof(char) * (read_length - 1U), &cbRead, NULL);
```

Reply:
Done for `ret` and `err`